src/basic/utf8.c

```diff
@@ -103,8 +103,9 @@ static int utf8_encoded_expected_len(const char *str) {
 }
 
 /* decode one unicode char */
-int utf8_encoded_to_unichar(const char *str) {
-        int unichar, len, i;
+int32_t utf8_encoded_to_unichar(const char *str) {
```

Comment:
To avoid further signed/unsigned confusion on the API layer, I'd much prefer to return the result in a pointer.